src/wallet/scriptpubkeyman.cpp

```diff
@@ -203,6 +198,16 @@ IsMineResult IsMineInner(const LegacyDataSPKM& keystore, const CScript& scriptPu
                 }
             }
         }
+        // Follow consensus rules, never treat too large legacy multisig scripts as valid
+        if (sigversion == IsMineSigVersion::P2SH && scriptPubKey.size() > MAX_SCRIPT_ELEMENT_SIZE) {
+            return IsMineResult::INVALID;
+        }
+
+        // Never treat bare multisig outputs as ours (they can still be made watchonly-though)
+        if (sigversion == IsMineSigVersion::TOP) {
+            if (keys.size() > 3) return IsMineResult::INVALID; // These are standard wise non-spendable
```

Comment:
nit: Maybe we should have a constant for this number (3)? Then we can use here and in `IsStandard`.

Reply:
yeah, thats also something that popped-up in (https://github.com/bitcoin/bitcoin/pull/31404#discussion_r1866127089). Will bring that commit here too.